### DISCUSSION TITLE
Hsts

### DISCUSSION THREAD

**Discussion by Notadumdum** [miguelgrinberg/python-socketio#1104] [Q&A] [answered]
Good day I just started reading about HSTS and I wanna know if it has any way possible from stopping a non-browser client from connecting to a socket.io server

**Comment by miguelgrinberg** [comment, marked as answer]
There is no way to distinguish a browser from a non-browser client. You can look at the user agent, but these are easily spoofed so this isn't a foolproof method. Not sure what HSTS article you read, but there is no relation between HSTS and detecting the type of client that connects to your server, regardless of being a Socket.IO server or just a plain HTTP server.

**Comment by Notadumdum** [comment]
Ok thank you @miguelgrinberg
